src/pages/Account/AdminEvent.tsx

```diff
@@ -0,0 +1,184 @@
+import React from "react";
+import { Modal } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import firebase from "firebase/app";
+import { VenueEvent } from "types/VenueEvent";
+import dayjs from "dayjs";
+import * as Yup from "yup";
+import isSameOrAfter from "dayjs/plugin/isSameOrAfter";
+dayjs.extend(isSameOrAfter);
+
+interface PropsType {
+  show: boolean;
+  onHide: () => void;
+  venueId: string;
+}
+
+interface CreateEventFormData {
+  name: string;
+  description: string;
+  start_date: string;
+  start_time: string;
+  end_date: string;
+  end_time: string;
+  price: number;
+}
+
+const postEvent = async (venueId: string, event: Omit<VenueEvent, "id">) => {
+  await firebase.firestore().collection(`venues/${venueId}/events`).add(event);
+};
+
+const validationSchema = Yup.object().shape<CreateEventFormData>({
+  name: Yup.string().required("Name required"),
+  description: Yup.string().required("Description required"),
+  start_date: Yup.string()
+    .required("Start date required")
+    .matches(
+      /\d{4}-\d{2}-\d{2}/,
+      'Start date must have the format "yyyy-mm-dd"'
+    )
+    .test(
+      "start_date_futur",
+      "Start date must be in the futur",
+      (start_date) => {
+        return dayjs(start_date).isSameOrAfter(dayjs(), "day");
+      }
+    ),
+  start_time: Yup.string().required("Start time required"),
+  end_date: Yup.string()
+    .required("End date required")
+    .test("end_date_futur", "End date must be in the futur", (end_date) => {
+      return dayjs(end_date).isSameOrAfter(dayjs(), "day");
+    }),
+  end_time: Yup.string().required("End time required"),
+  price: Yup.number()
+    .typeError("Price must be a number")
+    .required("Price is required"),
+});
+
+const AdminEvent: React.FunctionComponent<PropsType> = ({
+  show,
+  onHide,
+  venueId,
+}) => {
+  const { register, handleSubmit, errors, formState } = useForm<
+    CreateEventFormData
+  >({
+    mode: "onChange",
+    validationSchema,
+  });
+  const onSubmit = async (data: CreateEventFormData) => {
+    const start = dayjs(`${data.start_date} ${data.start_time}`);
+    const end = dayjs(`${data.end_date} ${data.end_time}`);
+    const event: Omit<VenueEvent, "id"> = {
```

Comment:
do we want 'id' being stored in firebase. if not, consider using lodash omit

Reply:
There is no `id` being set here. It's in the type because it's added by the hook the app uses.